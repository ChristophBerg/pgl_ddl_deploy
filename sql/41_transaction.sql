SET ROLE test_pgl_ddl_deploy;
SET client_min_messages TO warning;

BEGIN;

/***
In default schema
**/
CREATE TABLE foo(id serial primary key);
SELECT * FROM check_rep_tables();
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;

ALTER TABLE foo ADD COLUMN bla TEXT;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;

INSERT INTO foo (bla) VALUES (1),(2),(3);

DROP TABLE foo CASCADE;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;

CREATE TABLE foobar.foo(id serial primary key);
SELECT * FROM check_rep_tables();
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;

ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;

INSERT INTO foobar.foo (bla) VALUES (1),(2),(3);

DROP TABLE foobar.foo CASCADE;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;

COMMIT;

SELECT * FROM pgl_ddl_deploy.exceptions;
