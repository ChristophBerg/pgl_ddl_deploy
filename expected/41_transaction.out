SET ROLE test_pgl_ddl_deploy;
SET client_min_messages TO warning;
BEGIN;
/***
In default schema
**/
CREATE TABLE foo(id serial primary key);
SELECT * FROM check_rep_tables();
 set_name | table_name 
----------+------------
 test1    | foo
 test2    | foo
 test3    | foo
 test4    | foo
(4 rows)

SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;
 set_name |               ddl_sql_raw                |               ddl_sql_sent               
----------+------------------------------------------+------------------------------------------
 test4    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test3    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test2    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test1    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test8    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
 test7    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
 test6    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
 test5    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
 test4    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
 test3    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
(10 rows)

ALTER TABLE foo ADD COLUMN bla TEXT;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;
 set_name |               ddl_sql_raw                |               ddl_sql_sent               
----------+------------------------------------------+------------------------------------------
 test4    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test3    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test2    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test1    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test4    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test3    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test2    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test1    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test8    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
 test7    | DROP TABLE foobar.foo;                   | DROP TABLE foobar.foo;
(10 rows)

INSERT INTO foo (bla) VALUES (1),(2),(3);
DROP TABLE foo CASCADE;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;
 set_name |               ddl_sql_raw                |               ddl_sql_sent               
----------+------------------------------------------+------------------------------------------
 test4    | DROP TABLE foo CASCADE;                  | DROP TABLE foo CASCADE;
 test3    | DROP TABLE foo CASCADE;                  | DROP TABLE foo CASCADE;
 test2    | DROP TABLE foo CASCADE;                  | DROP TABLE foo CASCADE;
 test1    | DROP TABLE foo CASCADE;                  | DROP TABLE foo CASCADE;
 test4    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test3    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test2    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test1    | ALTER TABLE foo ADD COLUMN bla TEXT;     | ALTER TABLE foo ADD COLUMN bla TEXT;
 test4    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
 test3    | /***                                    +| /***                                    +
          | In default schema                       +| In default schema                       +
          | **/                                     +| **/                                     +
          | CREATE TABLE foo(id serial primary key); | CREATE TABLE foo(id serial primary key);
(10 rows)

CREATE TABLE foobar.foo(id serial primary key);
SELECT * FROM check_rep_tables();
 set_name | table_name 
----------+------------
 test1    | foobar.foo
 test2    | foobar.foo
 test3    | foobar.foo
 test4    | foobar.foo
 test5    | foobar.foo
 test6    | foobar.foo
 test7    | foobar.foo
 test8    | foobar.foo
(8 rows)

SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;
 set_name |                   ddl_sql_raw                   |                  ddl_sql_sent                   
----------+-------------------------------------------------+-------------------------------------------------
 test8    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test7    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test6    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test5    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test4    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test3    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test2    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test1    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test4    | DROP TABLE foo CASCADE;                         | DROP TABLE foo CASCADE;
 test3    | DROP TABLE foo CASCADE;                         | DROP TABLE foo CASCADE;
(10 rows)

ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;
 set_name |                   ddl_sql_raw                   |                  ddl_sql_sent                   
----------+-------------------------------------------------+-------------------------------------------------
 test8    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test7    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test6    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test5    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test4    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test3    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test2    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test1    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;     | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test8    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
 test7    | CREATE TABLE foobar.foo(id serial primary key); | CREATE TABLE foobar.foo(id serial primary key);
(10 rows)

INSERT INTO foobar.foo (bla) VALUES (1),(2),(3);
DROP TABLE foobar.foo CASCADE;
SELECT set_name, ddl_sql_raw, ddl_sql_sent FROM pgl_ddl_deploy.events ORDER BY id DESC LIMIT 10;
 set_name |                 ddl_sql_raw                 |                ddl_sql_sent                 
----------+---------------------------------------------+---------------------------------------------
 test8    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test7    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test6    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test5    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test4    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test3    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test2    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test1    | DROP TABLE foobar.foo CASCADE;              | DROP TABLE foobar.foo CASCADE;
 test8    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT; | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
 test7    | ALTER TABLE foobar.foo ADD COLUMN bla TEXT; | ALTER TABLE foobar.foo ADD COLUMN bla TEXT;
(10 rows)

COMMIT;
SELECT * FROM pgl_ddl_deploy.exceptions;
 id | set_name | pid | executed_at | ddl_sql | err_msg | err_state | set_config_id | resolved | resolved_notes 
----+----------+-----+-------------+---------+---------+-----------+---------------+----------+----------------
(0 rows)

